docs/hands-on-learning/introduction/module-3-recipe-development-environment.md

```diff
@@ -0,0 +1,88 @@
+---
+sidebar_label: "Module 3: Environment setup"
+description: How to set up your IDE to write custom recipes
+---
+
+# Module 3: Recipe development environment
+
+Now that you've seen how to run recipes and built one in Moderne using the recipe builder, let's look at how to write your own recipes.
+
+You'll want to have the following installed:
+
+* Java 21, as our [RewriteTests](https://docs.openrewrite.org/authoring-recipes/recipe-testing#rewritetest-interface) use text blocks.
+  * The [rewrite-recipe-starter](https://github.com/moderneinc/rewrite-recipe-starter) project expects Temurin JDK 21.0.7 (`temurin-21.0.7`).
+  * Recipes use Java 8 source level, so they can run on Java 8 and higher.
+* IntelliJ IDEA Ultimate 2024.1+ (required for the OpenRewrite plugin; Community Edition is not supported).
+* The [OpenRewrite plugin](https://plugins.jetbrains.com/plugin/23814-openrewrite), to run and write YAML recipes (This comes pre-installed with IntelliJ Ultimate versions 2024.1 or later).
+* [The Moderne plugin](../../user-documentation/moderne-ide-integration/how-to-guides/moderne-plugin-install.md), for faster recipe development and to help debug recipes.
+* [The Moderne CLI](../../user-documentation/moderne-cli/getting-started/cli-intro.md), to run recipes at scale locally, and debug against serialized LSTs.
+
+## Exercise 3: Create and test your own recipe module
```

Comment:
When I was reading through these modules, I found myself being like "hmm why is there an exercise 3 but not an exercise 2 or 1?"

I feel like for every one of these modules, we should remove the exercise number and just have it be like: "Exercise: <Description>"